src/index.ts

```diff
@@ -0,0 +1,543 @@
+          if (this.hasPollingSubscriptions) {
+            await this._restApi?.productInfo();
+          }
+          await this.pollClips();
+          await this.pollLayers();
+          this.status(this.STATUS_OK);
+        }
+        catch (e: any) {
+          this.status(this.STATUS_ERROR, e.message);
+        }
+        await sleep(500);
+      }
+      if (!this._restApi) {
+        // no way to tell if OSC is connected
+        this.status(this.STATUS_OK);
+      }
+    }
+    finally {
+      this.isPolling = false;
+    }
+  }
+
+  get hasPollingSubscriptions(): boolean {
+    return this.LayerBypassedSubscriptions.size > 0 ||
+      this.clipConnectedSubscriptions.size > 0 ||
+      this.clipStatusSubscriptions.size > 0;
+  }
+
+  async pollLayers() {
+    try {
+      if (this.LayerBypassedSubscriptions.size > 0) {
+        for (var layer of this.LayerBypassedSubscriptions) {
+          var status = (await this._restApi?.Layers.getSettings(layer)) as LayerOptions;
+          if (status.bypassed?.value) {
+            this.bypassedLayers.add(layer);
+          } else {
+            this.bypassedLayers.delete(layer);
+          }
+        }
+        this.checkFeedbacks('layerBypassed');
+      }
+    }
+    catch (e) {
+      console.log(e);
+    }
+  }
+
+  async pollClips() {
+    var clips = new Set<ClipSubscription>();
+    // combine sets
+    for (var clip of this.clipStatusSubscriptions) {
+      clips.add(clip);
+    }
+    for (var clip of this.clipConnectedSubscriptions) {
+      clips.add(clip);
+    }
+    if (clips.size > 0) {
+      let connectedChanged = false;
+      let nameChanged = false;
+      let thumbChanged = false;
+      for (var clip of clips) {
+        var key = `${clip.layer}-${clip.column}`;
+        var status = (await this._restApi?.Clips.getStatus(clip.layer, clip.column)) as ClipStatus;
+        var name = status?.name?.value;
+        this.clipStatus.set(key, status);
+        if (name !== this.clipNames.get(key)) {
+          this.clipNames.set(key, name);
+          nameChanged = true;
+        }
+        var isConnected = status?.connected.value === 'Connected';
+        if (isConnected) {
+          if (!this.connectedClips.has(key)) {
+            connectedChanged = true;
+            this.connectedClips.add(key);
+          }
+        } else {
+          if (this.connectedClips.has(key)) {
+            connectedChanged = true;
+            this.connectedClips.delete(key);
+          }
+        }
+        if (nameChanged && this.clipThumbSubscriptions.has(key)) {
+          this.clipThumbs.set(key, await this._restApi?.Clips.getThumb(clip.layer, clip.column) ?? '');
+          thumbChanged = true;
+        }
+      }
+      if (connectedChanged) {
+        this.checkFeedbacks('connectedClip');
+      }
+      if (nameChanged || thumbChanged) {
+        this.checkFeedbacks('clipInfo');
+      }
+    }
+
+  }
+}
+
+/*
+instance.GetUpgradeScripts = function() {
```

Comment:
You shouldn't remove this upgrade script. Is it not still useful for some users?

By removing it, when a new upgrade script is added later on, it wont get run for all users. The database of some users will say that its already run the first upgrade script, so wont re-run it.  
If it really isnt useful anymore, then make it be an empty function.